src/config.ts

```diff
@@ -88,18 +89,29 @@ export async function readDefaultConfig(root?: string): Promise<Config> {
   return normalizeConfig(await importConfig(tsPath), root);
 }
 
+let cachedPages: {key: string; pages: Page[]} | null = null;
+
 function readPages(root: string, md: MarkdownIt): Page[] {
-  const pages: Page[] = [];
+  const files: {file: string; source: string}[] = [];
+  const hash = createHash("sha256");
   for (const file of visitMarkdownFiles(root)) {
     if (file === "index.md" || file === "404.md") continue;
     const source = readFileSync(join(root, file), "utf8");
-    const parsed = parseMarkdown(source, {path: file, md});
+    files.push({file, source});
+    hash.update(file).update(source);
```

Comment:
I also tried hashing the mtime of the file instead of the source, which is slightly faster in the cached cache. But I don’t think it’s fast enough that it matters, and this felt simpler.